Python/Product/IronPython/Interpreter/IronPythonInterpreter.cs

```diff
@@ -126,7 +126,8 @@ class IronPythonInterpreter :
         }
 
 #if DEBUG
-        class AssertListenerInitializer {
+        [Serializable]
+        class AssertListenerInitializer : MarshalByRefObject {
```

Comment:
Misread the diff - I thought this was previously MarshalByRef.

The important thing to ensure here is that the listener is added within the appdomain. My understanding was that Serializable will achieve that (as the listener should be MarshalByRef), but marshalling the initializer will just add the listener in the original appdomain again.